### ISSUE TITLE
Failed to create standalone redis

### ISSUE THREAD

**Issue by ryanking8215** [bitnami/charts#22817]
### Name and Version

bitnami/redis-18.9.0

### What architecture are you using?

amd64

### What steps will reproduce the bug?
```
helm install redis  --set architecture=standalone oci://registry-1.docker.io/bitnamicharts/redis
```
### Are you using any custom parameters or values?
```
--set architecture=standalone 
```

### What is the expected behavior?

redis pods and service are created and run ok.

### What do you see instead?

No pods were created and serviceAccount is not created for standalone mode.
```console
$ k describe sts redis-master 
Warning  FailedCreate  2m22s (x17 over 7m50s)  statefulset-controller  create Pod redis-master-0 in StatefulSet redis-master failed error: pods "redis-master-0" is forbidden: error looking up service account infras/redis-master: serviceaccount "redis-master" not found
```

**Comment by opencmit2** [comment]
Hi @ryanking8215 
I haven't reproduced the problem you mentioned.
![image](https://github.com/bitnami/charts/assets/112474703/77818bbc-8488-44f0-b77a-dc1d5e8736ee)

**Comment by bjorngoossens** [comment]
Same here. Seems to happen after the 18.9.0 release. We deploy redis as standalone, and also get the same error when inspecting the StatefulSet. This is on DOKS with redis deployed through flux.

**Comment by mawatech** [comment]
I think this is a duplicate of https://github.com/bitnami/charts/issues/22746

**Comment by BogdanGeorge** [comment]
Same for me, is happens with latest version 18.9.0 even though i have serviceAccount create true in my config. With 18.8.2 works fine

**Comment by carrodher** [comment]
Are you able to reproduce the issue with the latest version (18.9.1)? It should be fixed at https://github.com/bitnami/charts/pull/22747

**Comment by ryanking8215** [comment]
Yes,  I tested and it is fixed with the last version (18.11.0).
